src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -294,10 +358,123 @@ public override async ValueTask<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(
                 Message = invokeRequest,
             };
 
-            return await this.MakeGrpcCallHandleError(
-                 options => client.InvokeServiceAsync(request, options),
-                 headers,
-                 cancellationToken);
+            var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+            // add token for dapr api token based authentication
+            var daprApiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+
+            if (daprApiToken != null)
+            {
+                callOptions.Headers.Add("dapr-api-token", daprApiToken);
+            }
+
+            return (request, callOptions);
+        }
+
+        private async Task<InvocationResponse<TResponse>> MakeInvokeRequestAsyncWithResponse<TRequest, TResponse>(
+            InvocationRequest<TRequest> request,
+            CancellationToken cancellationToken = default)
+        {
+            
+            
+            Any serializedData = null;
+            if (request.Body != null)
+            {
+                if (typeof(TResponse) == typeof(byte[]))
+                {
+                    // User has passed in raw bytes
+                    var requestBytes = (byte[])Convert.ChangeType(request.Body, typeof(byte[]));
```

Comment:
You don't need `Convert.ChangeType` here, just do a cast.

Reply:
I had tried the cast....getting this without the Convert.ChangeType. DaprClientGrpc.cs(377,40): error CS0030: Cannot convert type '(TRequest Body, System.Type)' to 'byte[]' [/Users/vinayada/dapr/dotnet-sdk/src/Dapr.Client/Dapr.Client.csproj]